timemachine/fe/atom_mapping.py

```diff
@@ -94,24 +95,39 @@ def get_core_by_mcs(mol_a, mol_b, query, threshold=0.5):
         the core indices can get flipped around,
         for example if the substructure match hits only part of an aromatic ring.
 
-        In some cases, this means that pairs of atoms that do not satisfy the
-        atom comparison function can be mapped together.
+        In some cases, this can fail to find a mapping that satisfies the distance
+        threshold, raising an AtomMappingError.
     """
 
     # fetch conformer, assumed aligned
     conf_a = mol_a.GetConformer(0).GetPositions()
     conf_b = mol_b.GetConformer(0).GetPositions()
 
     # note that >1 match possible here -- must pick minimum-cost match
-    matches_a = mol_a.GetSubstructMatches(query, uniquify=False)
-    matches_b = mol_b.GetSubstructMatches(query, uniquify=False)
+    max_matches = 10_000
```

Comment:
Are you matching with or without hydrogens?  I will probably be cheaper to match without hydrogens and then match the hydrogens after the fact.

Reply:
Attempted, and I think your suggestion would be faster, but I don't know how to implement the "match the hydrogens after the fact" step without making this function significantly more complicated or possibly returning matches that could be inconsistent with `query`.